### PR TITLE
refactor(iroh-net): generalize `derp` naming to `relay` to prepare for future refactors

### PR DESCRIPTION
**Pull request by ramfox**
## Description

We have been making breaking changes to the DERP protocol and are planning future changes that will better take advantage of / integrate with QUIC in our hole-punching schemes.

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.

### CODE REVIEW

File: .github/ansible/redeploy-derper.yml [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-      cmd: sudo setcap CAP_NET_BIND_SERVICE=+eip /usr/local/bin/derper
-  - name: Make sure derper is started
+      cmd: sudo setcap CAP_NET_BIND_SERVICE=+eip /usr/local/bin/iroh-relay
+  - name: Make sure iroh-relay is started
     ansible.builtin.systemd:
       state: restarted
       enabled: yes
-      name: derper
\ No newline at end of file
+      name: derper
```
Comment (Arqu) [inline .github/ansible/redeploy-derper.yml:20]: This should probably also be `iroh-relay` but will break regardless. Need to update infra once it lands. Will file an issue for myself.

File: .github/workflows/netsim.yml [hunk 2]
```diff
@@ -97,7 +97,7 @@ jobs:
     - name: Copy binaries to right location
       run: |
         cp target/optimized-release/iroh ../chuck/netsim/bins/iroh
-        cp target/optimized-release/derper ../chuck/netsim/bins/derper
+        cp target/optimized-release/iroh-relay ../chuck/netsim/bins/iroh-relay
```
Comment (Arqu) [inline .github/workflows/netsim.yml:100]: To cheat for now you can do `cp target/optimized-release/iroh-relay ../chuck/netsim/bins/derper` given commands/args have remained the same. I can do another pass to bring all the other infra in line with the rename.

File: .github/workflows/netsim.yml [hunk 3]
```diff
@@ -132,7 +132,7 @@ jobs:
       
     - name: Cleanup
       run: |
-        sudo kill -9 $(pgrep derper) || true
+        sudo kill -9 $(pgrep iroh-relay) || true
```
Comment (Arqu) [inline .github/workflows/netsim.yml:135]: cheat code `derper` instead of `iroh-relay`

**Review by Arqu — COMMENTED** [review]
Left some quick comments regarding the CI.

**Comment by Arqu** [comment]
The logs indicate that they fail to set up a connection (given it's mostly a rename PR, functionality should remain the same). I did a manual run and saw it trying to reach default derpers which those tests shouldn't. Which pointed me to https://github.com/n0-computer/chuck/blob/main/fixtures/derp.config.toml which we feed into the iroh binary at run time. 

To make things work with the current state of the PR we just need to rename `[[derp_nodes]]` into `[[relay_nodes]]` and it correctly picks up the config and works fine from there.

We can potentially roll this back to the first version you pinged me where we had everything renamed to `iroh-relay` and I can cook up a netsim branch that lines everything up with it. Then we just merge in succession.

File: .github/ansible/redeploy-derper.yml [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 
   tasks:
-  - name: Fetch derper binary
+  - name: Fetch iroh-relay binary
     get_url:
       url: https://vorc.s3.us-east-2.amazonaws.com/derper-linux-amd64-{{ derper_version }}
       mode: '0755'
       force: yes
-      dest: /usr/local/bin/derper
+      dest: /usr/local/bin/iroh-relay
```
Comment (Arqu) [inline .github/ansible/redeploy-derper.yml:12]: for the cheat code mode we still need this and te line bellow to be `derper` as I need to do the infra adjustments too.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -192,7 +192,7 @@ jobs:
     - name: Copy binaries to right location
       run: |
         cp target/release/iroh ../chuck/netsim/bins/iroh
-        cp target/release/derper ../chuck/netsim/bins/derper
+        cp target/release/iroh-relay ../chuck/netsim/bins/derper
```
Comment (Arqu) [inline .github/workflows/ci.yml:195]: 👍 this is good

File: .github/workflows/test_relay_server.yml [hunk 6]
```diff
@@ -50,18 +50,18 @@ jobs:
...
           run: |
               echo ::set-output name=tag::${GITHUB_SHA::7}
         
 
-    deploy_derper:
+    deploy_iroh-relay:
         runs-on: ubuntu-latest
         if: github.ref_name=='main'
-        needs: build_derper
+        needs: build_iroh_relay
```
Comment (Arqu) [inline .github/workflows/test_relay_server.yml:64]: this should be `build_relay_server` as the job above

**Review by Arqu — COMMENTED** [review]
And we need to update the release ci script

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**